include/openssl/params.h

```diff
@@ -130,6 +130,51 @@ int OSSL_PARAM_get_octet_ptr(const OSSL_PARAM *p, const void **val,
 int OSSL_PARAM_set_octet_ptr(OSSL_PARAM *p, const void *val,
                              size_t used_len);
 
+#define OSSL_PARAM_TYPE_int 1
+#define OSSL_PARAM_TYPE_uint 2
+#define OSSL_PARAM_TYPE_long 3
+#define OSSL_PARAM_TYPE_ulong 4
+#define OSSL_PARAM_TYPE_int32 5
+#define OSSL_PARAM_TYPE_uint32 6
+#define OSSL_PARAM_TYPE_int64 7
+#define OSSL_PARAM_TYPE_uint64 8
+#define OSSL_PARAM_TYPE_size_t 9
+#define OSSL_PARAM_TYPE_double 10
+#define OSSL_PARAM_TYPE_BN 11
+#define OSSL_PARAM_TYPE_utf8 12
+#define OSSL_PARAM_TYPE_octet 13
+
+OSSL_PARAM *OSSL_PARAM_build(const char *name, ...);
+void OSSL_PARAM_build_free(OSSL_PARAM *params);
+
+#define OSSL_PARAM_BUILD_int(name, value) \
```

Comment:
Hmmm, the manual says "address", not "value". Please decide.

Reply:
Name for scalar types, address for strings and BN's.